lib/common/src/Dom/Document.php

```diff
@@ -1609,6 +1610,18 @@ public function __get($name)
                     $this->body = $this->getElementsByTagName(Tag::BODY)->item(0);
                 }
                 return $this->body;
+            case 'viewport':
+                // TODO: Can this be cached in an instance property as well?
```

Comment:
I suppose the only concern would be that someone could remove the viewport after it was first retrieved. It could check to see if it is still a child of `$this->head` and then return, or else do another search.

Reply:
I'm not so much worried about it changing after the fact, as I am that it could then be queried too early when it hasn't yet been added. If that is then stored, even adding it later won't fix. I think this is quite likely to happen, as it is not part of the "enforced normalized structure", but is added a bit later after the normalization has already finished.

I'll leave this dynamic for now.